src/App.js

```diff
@@ -1,19 +1,27 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [totalLikes, setTotalLikes] = useState(0);
+
+  const handleLikeCount = (id, liked) => {
+    setTotalLikes((prevLikes) => (liked ? prevLikes + 1 : prevLikes - 1));
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Vladimir and Estragon</h1>
+        <p>{totalLikes} {totalLikes === 1 ? '❤️' : '❤️s'}</p>
       </header>
       <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
+        <ChatLog entries={chatMessages} onLikeChange={handleLikeCount} />
```

Comment:
👀 The array of message data contains information about whether each message is liked, but unless we store the messages in a piece of state and update it in response to user clicks, we won't be able to make use of the liked value in displaying our UI. So rather than having liked state tracked down in ChatEntry per message, we would like to bring the responsibility for managing that state up to the App level, and pass down the current message state to the ChatLog here.

To help motivate this, imagine that the message data was stored in a backend database (similar to the tasks in task list, or the cards in inspiration board). When the liked status changes, we really want to reflect that liked status change in the database, not only in the UI. While we don't have a backend here, we can get as close to that as possible by lifting the state management up to a list of messages here in App.